test/integration/graphql_test.lua

```diff
@@ -1158,4 +1158,43 @@ function g.test_both_data_and_error_result()
         {message = 'Simple error A'},
         {message = 'Simple error B'},
     })
+
+    query = [[{
+        prefix {
+            test_A: test(arg: "A")
+            test_B: test(arg: "B")
+        }
+    }]]
+
+    query_schema = {
+        ['prefix'] = {
+            kind = {
+                __type = 'Object',
+                name = 'prefix',
+                fields = {
```

Comment:
I think it could be rewritten using `type.Object` function. Currently it seems you use info about internals here.